### PR TITLE
Fix tests on cs-CZ culture

### PR DESCRIPTION
**Pull request by svick**
**Customer scenario**

When I run tests for Roslyn on my machine with cs-CZ culture and Czech default encoding, some tests fail.

**Bugs this fixes:**

None.

**Workarounds, if any**

Switching to the en-US culture fixes some of these issues.

**Risk**

This only changes tests, so risk should be small.

**Performance impact**

This only changes tests, so performance of the compiler won't be affected.

**Is this a regression from a previous update?** No.

**Root cause analysis:**

How did we miss it? By only running tests on en-US culture and English default encoding.

What tests are we adding to guard against it in the future? I don't think tests for tests are necessary.

**How was the bug found?**

By me running `Test.cmd` on my machine.

### CODE REVIEW

File: src/Compilers/Core/CodeAnalysisTest/Text/StringTextDecodingTests.cs [hunk 1]
```diff
@@ -133,7 +133,7 @@ public void CheckSum_SHA256()
             Assert.Equal("f1945cd6 c19e56b3 c1c78943 ef5ec181 16907a4c a1efc40a 57d48ab1 db7adfc5", StringTextTest.ChecksumToHexQuads(checksum));
         }
 
-        [ConditionalFact(typeof(IsEnglishLocal))]
+        [ConditionalFact(typeof(HasEnglishDefaultEncoding))]
         [WorkItem(5663, "https://github.com/dotnet/roslyn/issues/5663")]
         public void Decode_NonUtf8()
```
Comment (svick) [inline src/Compilers/Core/CodeAnalysisTest/Text/StringTextDecodingTests.cs]: This test failed on my machine, even though both `CurrentCulture` and `CurrentUICulture` returned `un-US`. It seems the default encoding (which is Windows-1250, not Windows-1252, on my machine) is not the same as current culture.

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 2]
```diff
@@ -845,8 +859,7 @@ public override void VisitLiteralExpression(ILiteralExpression operation)
         {
             LogString(nameof(ILiteralExpression));
 
-            object value;
-            if (operation.ConstantValue.HasValue && ((value = operation.ConstantValue.Value) == null ? "null" : value.ToString()) == operation.Text)
```
Comment (svick) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: I don't understand this line. It means the raw text of the node is printed only when it's equal to calling `ToString()` on the `object` value. Shouldn't it be the other way around, that the raw text is printed when it contains something unexpected?

@mavasani As far as I can tell, you introduced this line [here](https://github.com/dotnet/roslyn/pull/11918/files?diff=unified#diff-0926b7ba2553abfab6093f42cf274213R793). Do you happen to remember why it's written like this?

Anyway, this PR doesn't change the logic, only makes it use the invariant culture. This means existing tests still pass.

**Comment by cston** [comment]
LGTM

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 3]
```diff
@@ -163,13 +164,26 @@ private void LogConstant(Optional<object> constant, string header = "Constant")
...
-            var valueStr = constant != null ? constant.ToString() : "null";
-            if (constant is string)
+            switch (constant)
             {
-                valueStr = @"""" + valueStr + @"""";
+                case null:
+                    return "null";
+                case string s:
+                    if (quoteString)
+                        return @"""" + s + @"""";
```
Comment (jcouv) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: Style: braces are requires in the compiler.

Comment (svick) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: @jcouv Fixed.

**Comment by svick** [comment]
windows_debug_vs-integration_prtest failed because of timeout. I don't see how this PR could have caused that, is it a flaky test?

**Comment by jcouv** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest  please

**Pull request merged**